From caaa78c341659de66cf478402dbf302169e04f23 Mon Sep 17 00:00:00 2001
From: Yann Gautier <yann.gautier@foss.st.com>
Date: Tue, 7 Nov 2023 18:33:45 +0100
Subject: [PATCH 1040/1695] plat-stm32mp2: conf: add CFG_STM32_CM33TDCID

If Cortex-M33 is TDCID, enable a dedicated flag CFG_STM32_CM33TDCID.
This can be enabled automatically if the board DT is in the
flavorlist-M33-TDCID.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git 0cbb1234d274a54645c74f962bf1b173dabda5a9 plat-stm32mp2: conf: add CFG_STM32_CM33TDCID]
Change-Id: I5efc9e58d82aa807380c9c11d6488475d2b68409
Signed-off-by: Yann Gautier <yann.gautier@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/397749
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
Reviewed-by: Gatien CHEVALLIER <gatien.chevallier@st.com>
Tested-by: Gatien CHEVALLIER <gatien.chevallier@st.com>
Reviewed-by: Thomas BOURGOIN <thomas.bourgoin@foss.st.com>
---
 core/arch/arm/plat-stm32mp2/conf.mk | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/core/arch/arm/plat-stm32mp2/conf.mk b/core/arch/arm/plat-stm32mp2/conf.mk
index a2f6f7318..ada8197eb 100644
--- a/core/arch/arm/plat-stm32mp2/conf.mk
+++ b/core/arch/arm/plat-stm32mp2/conf.mk
@@ -79,6 +79,12 @@ endif
 include core/arch/arm/cpu/cortex-armv8-0.mk
 supported-ta-targets ?= ta_arm64
 
+# Enable flag when Cortex-M33 is TDCID
+ifneq ($(filter $(CFG_EMBED_DTB_SOURCE_FILE),$(flavorlist-M33-TDCID)),)
+$(call force,CFG_STM32_CM33TDCID,y)
+endif
+CFG_STM32_CM33TDCID ?= n
+
 $(call force,CFG_ARM_GIC_PM,y)
 $(call force,CFG_ARM64_core,y)
 $(call force,CFG_CORE_ASYNC_NOTIF,y)
-- 
2.47.2

